ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/trino/sampler.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Helper module to handle data sampling
+for the profiler
+"""
+from sqlalchemy import and_, inspect, text
+
+from metadata.generated.schema.entity.data.table import TableData
+from metadata.profiler.orm.registry import FLOAT_SET
+from metadata.profiler.processor.handle_partition import RANDOM_LABEL
+from metadata.profiler.processor.sampler.sqlalchemy.sampler import SQASampler
+
+
+class TrinoSampler(SQASampler):
+    """
+    Generates a sample of the data to not
+    run the query in the whole table.
+    """
+
+    def fetch_sample_data(self) -> TableData:
```

Comment:
I think what we need to override is not this method but `get_sample_query()`. Maybe `get_sample_query()` can be optimize to allow other sampler to just override the changing part (i.e. the queries and not the whole method)